### PR TITLE
feat(UI) : implement the timeline graph

### PR DESCRIPTION
**Pull request by Med1233**
## Description

 implement the timeline graph 

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [ ] master

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po
Comment (Thebarda) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:1]: Add translations on other languages

File: centreon/packages/ui/cypress/visual-testing-snapshots/src/Graph/Timeline/Timeline.cypress.spec.tsx/Timeline -- displays tooltip with correct information when hovered over a bar.snap.png
Comment (Thebarda) [inline centreon/packages/ui/cypress/visual-testing-snapshots/src/Graph/Timeline/Timeline.cypress.spec.tsx/Timeline -- displays tooltip with correct information when hovered over a bar.snap.png:1]: Add a small shadow on the tooltip. The fourth elevation should be enough

File: centreon/packages/ui/src/Graph/Timeline/timeline.styles.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { makeStyles } from 'tss-react/mui';
+
+export const useStyles = makeStyles()((theme) => ({
+  tooltip: {
+    backgroundColor: theme.palette.background.paper,
+    color: theme.palette.text.primary,
+    padding: 0,
+    position: 'relative',
+    left: theme.spacing(-1)
```
Comment (Thebarda) [inline centreon/packages/ui/src/Graph/Timeline/timeline.styles.ts:9]:
```suggestion
    left: theme.spacing(-1),
    boxShadow: theme.shadows[3]
```

**Comment by Thebarda** [comment]
There is an error in your story: https://65539c26347a2b106b5c65a9-jraellxpae.chromatic.com/?path=/story/graph-timeline--normal&globals=viewport:tablet;viewportRotated:!true

File: centreon/packages/ui/.storybook/preview.tsx [hunk 2]
```diff
@@ -9,6 +9,7 @@ import { allModes } from './modes';
 import { Decorator, Preview } from "@storybook/react";
 import { ReactQueryDevtools } from "@tanstack/react-query-devtools";
 
+
```
Comment (Med1233) [inline centreon/packages/ui/.storybook/preview.tsx:12]:
```suggestion
```

**Comment by Thebarda** [comment]
Screenshot 2024-10-02 at 09 20 31
@Wedierre @Med1233  Why is the tooltip displayed at the right/left of the anchor? It covers most of the timeline. Could it be displayed at the top of anchor?

**Comment by Med1233** [comment]
> Screenshot 2024-10-02 at 09 20 31 @Wedierre @Med1233 Why is the tooltip displayed at the right/left of the anchor? It covers most of the timeline. Could it be displayed at the top of anchor?

@Thebarda  This is not the actual tooltip for the BA timeline widget. It is only for testing purposes to ensure that we receive the necessary information

File: centreon/packages/ui/src/Graph/Timeline/ResponsiveTimeline.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                end={format({
+                  date: dayjs(end).tz(timezone).toDate(),
+                  formatString: dateTimeFormat
+                })}
+                color={color}
+                duration={getTimeDifference(dayjs(start), dayjs(end))}
+              />
+            )
+          }
+          position="right"
```
Comment (Thebarda) [inline centreon/packages/ui/src/Graph/Timeline/ResponsiveTimeline.tsx:77]:
```suggestion
          position="top"
```

**Review by Yassir-BenBOUBKER — APPROVED** [review]

**Review by quborg — APPROVED** [review]

**Comment by Wedierre** [comment]
> Screenshot 2024-10-02 at 09 20 31 @Wedierre @Med1233 Why is the tooltip displayed at the right/left of the anchor? It covers most of the timeline. Could it be displayed at the top of anchor?

Indeed it should not overlay the timeline and hide the content.

**Review by Wedierre — APPROVED** [review]

**Pull request merged**